scalyr_agent/configuration.py

```diff
@@ -223,6 +223,53 @@ def parse(self):
                 if https_server != server:
                     self.__config["scalyr_server"] = https_server
 
+            # Set defaults based on `max_send_rate_enforcement` value
+            if (
+                self.__config["max_send_rate_enforcement"] == "legacy"
+                or self.__config["disable_max_send_rate_enforcement_overrides"]
+            ):
+                if "max_allowed_request_size" not in self.__config:
+                    self.__config["max_allowed_request_size"] = 1024 * 1024
+                if "pipeline_threshold" not in self.__config:
+                    self.__config["pipeline_threshold"] = 1.1
+                if "min_request_spacing_interval" not in self.__config:
+                    self.__config["min_request_spacing_interval"] = 1.0
+                if "max_request_spacing_interval" not in self.__config:
+                    self.__config["max_request_spacing_interval"] = 5.0
+                if "max_log_offset_size" not in self.__config:
+                    self.__config["max_log_offset_size"] = 5 * 1024 * 1024
+                if "max_existing_log_offset_size" not in self.__config:
+                    self.__config["max_existing_log_offset_size"] = 100 * 1024 * 1024
+            else:
+                if "max_allowed_request_size" not in self.__config:
```

Comment:
Sorry if it wasn't clear from the spec, but these values (like `5900000`) should be used no matter what, even if the customer has some value defined in the config.

And yes, thinking about it, calling it "overriding defaults" is misleading in this way.  My apologies for that.

So, you shouldn't be doing any of the `not in self.__config` checks.